### PR TITLE
WIP: Feat/helm chart

### PR DESCRIPTION
**Pull request by gmoirod**
This is a Helm Chart to easily deploy WorkAdventure on K8S. Actually for testing purposes
Started with "helm create" so define lot of values not used by default.
There is some things left to do.
Open PR to share with interested persons.

### CODE REVIEW

**Comment by moufmouf** [comment]
Hey @gmoirod ,

Thanks a lot for this contribution. I really appreciate the hard work you put into this.

To be completely honest, I have mixed feelings about merging this in the main repository:

- I understand a Helm chart can be useful
- The maintainers of WorkAdventure don't have the skills to maintain it (I have played with Helm a few years ago and it was really a dreadful experience to me)
- Actually I hated so much Helm that I started working on [Deeployer](https://github.com/thecodingmachine/deeployer/), a tool that makes deploying in Kubernetes easier (and we are using it to deploy our test environments (see https://github.com/thecodingmachine/workadventure/blob/develop/deeployer.libsonnet)
- But Deeployer is nowhere near to be ready for production usage

So basically, I believe this Helm chart can definitely be useful, but I don't think we have the skills nor will to maintain it.

What do you think about this:

1- Deploy this Helm chart in another repository (this can be yours, or if you want, we can share membership of the project)
2- We add a link to your Helm repository in the documentation

**Comment by gmoirod** [comment]
Hey,

I understand your thoughts about Helm. 
In the meantime, i checked at your package manager deeployer and i doubt about it. If the file descriptor is simple, the complexity must be present in the runtime. Whereas in Helm, the complexity is in the templates....

I have no worries to maintain my own repo. You will find it here : https://github.com/gmoirod/helm-workadventure

**Comment by abbi-gaurav** [comment]
Hi @gmoirod , I am exploring your changes for a helm chart and used them to deploy on a kubernetes cluster with istio installed (it is open source project Kyma if you know).
So instead of ingress, I have istio virtual services that are exposing front, pusher, maps, uploader etc.

I am seeing the following errors. Any inputs on how can I resolve them?
- "Invalid room ID" error. 
- The GET call fails with 404 `(index):311 GET https://www.googletagmanager.com/gtm.js?id=GTM-KG975ZQ net::ERR_ABORTED 404`
- This one with BAD request `sa.js?key=7hknpcr7dkestq7ll1ot3:7 GET https://in-automate.sendinblue.com/p?key=7hknpcr7dkestq7ll1ot3&cuid=&ma_url=https%3A%2F%2Fwww.thecodingmachine.com%2Fbienvenue-chez-the-coding-machine%2F&sib_type=page&ma_title=Bienvenue%20chez%20The%20Coding%20Machine%20!%20-%20TheCodingMachine&sib_name=Bienvenue%20chez%20The%20Coding%20Machine%20!%20-%20TheCodingMachine&ma_referrer=<path to my front>&ma_path=%2Fbienvenue-chez-the-coding-machine%2F 400 (Bad Request)`

Thanks,
Gaurav

**Comment by gmoirod** [comment]
Hi @abbi-gaurav,

I don't know which version of workaventure docker images you deployed but a recent version could be the main cause. 

What I can say :
- 1st point is known. I think this is because api normally manages private rooms. But the public version of workadventure (free and open-source) does not provide this feature. This is not a blocking problem to get public universes to work (urls with a _). 

- The 2 latter points are unknown to me. But a quick search lead to an analytic service and a mailing service. I suppose workadventure introduced this to get metrics about usage and a way to provide a newsletter subscription in some way... In my opinion, this should be non blocking for the main app to work. 

You did not precise if the errors come from client browser or a workadventure component. But again I cannot figure why such errors would be blocking. 

Regards

**Comment by abbi-gaurav** [comment]
Thanks @gmoirod ,
I see the errors in the client browser.

**Comment by campey** [comment]
Looking at this: https://github.com/workadventure/workadventure/blob/develop/contrib/helm/README.md
It looks like there is now a working helm chart, can we close this PR?
